.github/workflows/wp-coding-standards.yml

```diff
@@ -0,0 +1,44 @@
+name: WordPress Coding Standards
+
+on:
+  push:
+    branches: [ trunk, develop ]
+  pull_request:
+    branches: [ trunk ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        php: [ 8.2 ]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup proper PHP version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer
+
+      - name: Validate composer.json and composer.lock
+        run: composer validate --strict
+
+      - name: Get composer cache directory
+        id: composer-cache
+        run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-
+
+      - name: Install dependencies
+        run: composer run-script packages-install
+
+      - name: Run PHP_CodeSniffer
+        run: composer run-script lint:php
```

Comment:
I'm noticing that the linting standard is left to a .phpcs.xml file that's embedded in each part of the project. This is great because it adds flexibility to each project, but my concern is that it makes global rule changes accross our repos difficult to maintain.

Is there anything we can do to increase cross-repo maintenance with a change like this?

Reply:
> Is there anything we can do to increase cross-repo maintenance with a change like this?

Actually, yes! When I reworked the `.phpcs.xml.dist` file, I suggested creating a repository like `a8cteam51/scripts` (or something along those lines) where we could store default configurations for PHPCS and maybe other stuff. Then we could just include it as a Composer package and link it just as the `themes/<theme>/.phpcs.xml` file links the config one at the root level.

Overall, the `.phpcs.xml` files embedded in different parts of the project are mostly meant to handle language domain checks and that global functions/variables are prefixed properly. And that's _really_ project-specific.